src/google_apps_clj/google_sheets_v4.clj

```diff
@@ -64,8 +65,34 @@
       (.get spreadsheet-id)
       (.execute)))
 
+(defn get-sheet-titles
+  [service spreadsheet-id]
+  "returns a list of [sheet-title sheet-id] tuples. It seems the order reflects
+  the order of the tabs in google's interface, though I doubt this is anywhere
+  guaranteed."
+  (->> (get (get-sheet-info service spreadsheet-id) "sheets")
+       (map #(get % "properties"))
+       (map (juxt #(get % "title") #(get % "sheetId")))))
```

Comment:
I generally prefer to provide fully realized, not lazy, seqs from impure fns, on the principle that I want to find out about problems from the sad angry outside world at the point of invocation, not farther afield in realization.